components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -136,9 +141,27 @@ void DistillPageText(
       base::UTF16ToUTF8(base::JoinString(text_node_contents, u" "));
 
   if (contents_text.empty()) {
-    std::move(callback).Run({});
+    blink::WebScriptSource source = blink::WebScriptSource(
+        blink::WebString::FromASCII("document.body.innerText"));
+    v8::HandleScope handle_scope(
+        render_frame->GetAgentGroupScheduler().Isolate());
+    v8::Local<v8::Value> result =
+        render_frame->GetWebFrame()->ExecuteScriptAndReturnValue(source);
+
+    if (result.IsEmpty() && !result->IsString()) {
+      std::move(callback).Run({});
+      return;
+    }
+
+    std::unique_ptr<base::Value> contents =
+        content::V8ValueConverter::Create()->FromV8Value(
+            result.As<v8::String>(),
```

Comment:
cannot use below because of a dep issue that gn_check complains

`String contents = blink::ToBlinkString<String>(result.As<v8::String>(), blink::kExternalize);`